executor/src/core.rs

```diff
@@ -143,54 +147,175 @@ where
                 // the second batch since there is no point in executing twice the same transactions
                 // (as the second execution attempt will always fail).
                 debug!("Duplicate batch {batch_digest}");
-                self.execution_indices.skip_batch(total_batches);
-                return Ok(());
+                return Ok(Vec::new());
             }
         };
 
+        // The consensus simply orders bytes, so we first need to deserialize the transaction.
+        // If the deserialization fail it is safe to ignore the transaction since all correct
+        // clients will do the same. Remember that a bad authority or client may input random
+        // bytes to the consensus.
+        let transactions = transactions
+            .into_iter()
+            .map(
+                |serialized| match bincode::deserialize::<State::Transaction>(&serialized) {
+                    Ok(x) => Ok(x),
+                    Err(e) => Err(SubscriberError::ClientExecutionError(format!(
+                        "Failed to deserialize transaction: {e}"
+                    ))),
+                },
+            )
+            .collect::<Result<Vec<_>, _>>()?;
```

Comment:
Could we collect a Vec<Result> rather than a Result<Vec> so that we can skip faulty transactions rather than the whole batch?

Reply:
I thought about this while debugging one of the unit tests that failed to deserialise (the compiler decided to send `i32` instead of `u64`). I think this is a bit of a grey area. I see that you label the error as `ClientExecutionError` which, during execution, is treated as a non-fatal one. However, it's not clear whether treating deseralisation errors as non-fatal would be correct. 

That's because it's difficult to say why we can't deserialise a transaction: Is it because some malicious validator put a malformed message into one of the batches? Or is it because the `Transaction` type on our side is somehow incorrect, and our machine is the only one that fails to deserialize something? Perhaps other validators have added an extra variant to an enum, but we forgot to update our node? Skipping such transactions could lead to consensus failure down the road. 

What we should do depends on how much pre-validation happens on the contents _before_ quorum is reached over their availability. Can malicious validators use Narwhal to atomically broadcast absolute rubbish content? Or is there some vetting before we vote on it to see if it at least conforms to some basic expectations about format?

If there is such validation, then we could be sure that the honest majority thinks this transaction looks legit, so if we can't deserialize it, we should stop and fix our software. If there is no such validation, then we can't decide whose fault is it, ours or the batcher's, in which case I don't know what to do.

With that in mind, do you want to just issue a warning and skip?

Reply:
Changed it back to skipping.

Reply:
On further thought, changed it to return raw bytes at this stage.

Reply:
It's a good point. Validators can input any rubbish as transactions, there are no checks on transactions format upon voting. Before voting, validators simply verify that the payload is available (and that the header doesn't break any safety or GC rules).

You are right that if the transaction serialisation format changes and some validators are late to the party, then it may be a safety problem. Should we open an issue for that and see what others think about it?

Reply:
Opening an issue can't hurt, at least there would be a story about it it. Someone using this library might want to protect themselves from the clients sending invalid content. If it's encoded into a trait, anyone happy with raw bytes can trivially allow everything in.